src/main/java/gift/controller/AuthController.java

```diff
@@ -0,0 +1,71 @@
+package gift.controller;
+
+import gift.dao.UserDAO;
+import gift.dto.AuthRequest;
+import gift.dto.AuthResponse;
+import gift.entity.User;
+import gift.exception.EmailAlreadyExistsException;
+import gift.exception.UnauthorizedException;
+import gift.exception.UserAuthException;
+import gift.service.TokenService;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AuthController {
+
+    private final TokenService tokenService;
+    private final UserDAO userDao;
+
+    @Autowired
+    public AuthController(TokenService tokenService, UserDAO userDao) {
+        this.tokenService = tokenService;
+        this.userDao = userDao;
+    }
+
+    @PostMapping("/members/login")
+    public ResponseEntity<?> login(@RequestBody AuthRequest authRequest) {
+        User user = userDao.findByEmail(authRequest.getEmail());
+
+        if (user == null || !user.getPassword().equals(authRequest.getPassword())) {
```

Comment:
user에서 getPassword()로 데이터를 꺼내서 다른 값과 equals 비교하는 대신, user 내부에서 동일한 패스워드인지 확인해보는건 어떨까요?  

```suggestion
        if (user == null || !user.samePassword(authRequest.getPassword()) {
```


```
규칙 9: getter/setter/프로퍼티를 쓰지 않는다.
```
위 항목에 대해 생각해보면 좋을 것 같아요.

Reply:
entity에서의 직접적인 필드값 조정이나 get을 하지 말라는 뜻으로 규칙 9를 해석하면 될까요?

또한 Entity 내부에서 직접 samePassword 메서드를 작성하는 것보다 UserService에서 해당 역할을 수행하는 메서드를 넣고 컨트롤러에 UserService 의존성을 주입하는 방식으로 사용하는 것에 대해서는 어떻게 생각하시는지 궁금합니다!